forge-game/src/main/java/forge/game/GameAction.java

```diff
@@ -131,31 +131,19 @@ private Card changeZone(final Zone zoneFrom, Zone zoneTo, final Card c, Integer
         // need to check before it enters
         if (c.isAura() && !c.isAttachedToEntity() && toBattlefield && (zoneFrom == null || !zoneFrom.is(ZoneType.Stack))) {
             boolean found = false;
-            try {
-                if (Iterables.any(game.getPlayers(), PlayerPredicates.canBeAttached(c, null))) {
-                    found = true;
-                }
-            } catch (Exception e1) {
-                found = false;
+            if (Iterables.any(game.getPlayers(), PlayerPredicates.canBeAttached(c, null))) {
+                found = true;
             }
 
             if (!found) {
-                try {
-                    if (Iterables.any((CardCollectionView) params.get(AbilityKey.LastStateBattlefield), CardPredicates.canBeAttached(c, null))) {
-                        found = true;
-                    }
-                } catch (Exception e2) {
-                    found = false;
+                if (Iterables.any((CardCollectionView) params.get(AbilityKey.LastStateBattlefield), CardPredicates.canBeAttached(c, null))) {
```

Comment:
maybe we can/should just reuse the `lastStateBattlefield` from above here?
what would a good test for this be 🤔

Reply:
yeah, better reuse the lastStateBattlefield that already get set

your current change with removing the try-catch would otherwise crash on possible null-pointer?

Reply:
there used to be an old crash, which I failed to reproduce in newer versions
I think at this point all relevant calls already provided the lastState
And if they didn't there could be a bug hidden because we might miss out on a legal attach
Anyway with the additional change it should now be impossible